<launch>
    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="L_a" type="double" value="0.125" />
    <param name="L_b" type="double" value="0.125" />
    <param name="m" type="double" value="1.98" />
    <param name="I_z" type="double" value="0.03" />
    
    <!-- flags -->
    <param name="sim_flag" type="bool" value="true" />
    <param name="feature_flag" type="bool" value="false" />

    <!-- Track parameters -->
    <param name="trackShape" type="str" value="oval" />

    <!-- Machines -->
    <machine name="odroid" address="192.168.10.213" user="odroid" timeout="20.0" env-loader="/home/odroid/barc/env_loader_odroid_ugo.sh"/>

    <!-- <machine name="this-pc" address="10.0.0.72" timeout="20.0" env-loader="/home/mpcubuntu/barc/env_loader_pc.sh"/> -->
    <machine name="this-pc" address="192.168.10.157" user="ugo" timeout="20.0" env-loader="/home/mpcubuntu/barc/env_loader_pc.sh"/>
    
    <!-- START NODES -->

    <!-- Estimator -->
    <node machine="this-pc" pkg="barc" type="stateEstimator.py" name="state_estimator" output="screen">
    <!-- delay in state estuiamtor -->
        <param name="delay_df" type="double" value="0.0" />
        <param name="delay_a" type="double" value="0.0" />
        <!-- ekf properties -->
        <param name="Q_x" type="double" value="0.0" />
        <param name="Q_y" type="double" value="0.0" />
        <param name="Q_vx" type="double" value="0.0" />
        <param name="Q_vy" type="double" value="0.0" />
        <param name="Q_ax" type="double" value="1.0" />
        <param name="Q_ay" type="double" value="1.0" />
        <param name="Q_psi" type="double" value="0.0" />
        <param name="Q_psiDot" type="double" value="1.0" />
        <param name="R_x" type="double" value="10.0" />
        <param name="R_y" type="double" value="10.0" />
        <param name="R_vx" type="double" value="0.1" />
        <param name="R_vy" type="double" value="0.01" />
        <param name="R_ax" type="double" value="10.0" />
        <param name="R_ay" type="double" value="10.0" />
        <param name="R_psiDot" type="double" value="0.1" />
    </node>

    <!-- LMPC CONTROL -->
    <node machine="this-pc" pkg="barc" type="controllerMain.py" name="control" output="screen">
        <param name="mode" type="str" value="simulations" />
    </node>

    <!-- LOW LEVEL CONTROLLER -->
    <node machine="this-pc" pkg="barc" type="controllerLowLevel.py" name="low_level_controller" />

    <!-- VISUALIZATION -->
    <node machine="this-pc" pkg="barc" type="plotCarTrajectory.py" name="visualization" />

    <!-- IMU -->
    <node machine="odroid" pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
    </node>

    <!-- GPS -->
    <node machine="odroid" pkg="marvelmind_nav" type="hedge_rcv_bin" name="indoor_gps" args="/dev/ttyACM1"/>
    
    <!-- ARDUINO -->
    <!--  * encoders and ultrasound sensors -->
    <node machine="odroid" pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node>

</launch>
